docs: guide on denoflare serve

marshallswain opened the pull request:
This adds a guide on how to get started building with Denoflare.

Closes https://github.com/skymethod/denoflare/issues/1

![Screen Shot 2021-11-20 at 11 04 10 PM](https://user-images.githubusercontent.com/128857/142751612-eee957d5-5c7d-454a-9651-f456b96eb4b0.jpg)

marshallswain committed:
docs: guide on denoflare serve

This adds a guide on how to get started building with Denoflare.

Closes https://github.com/skymethod/denoflare/issues/1

johnspurlock-skymethod commented on guides/serve.md:
```diff
@@ -0,0 +1,60 @@
+    {
+        "$schema": "https://raw.githubusercontent.com/skymethod/denoflare/v0.3.3/common/config.schema.json",
+        "scripts": {
+            "local": {
```
Maybe "hello-world-local" or "hello-local"?  I'm afraid "local" makes it look like a category or designation for all local scripts, rather than simply an identifier

marshallswain replied:
Oh good point.  I'll update that now

johnspurlock-skymethod commented:
This is fantastic, thank you!  Just one quibble re: the example script identifier "local" above.

marshallswain commented:
Ok. I've renamed the script and the part that referenced the script name.  Thanks for this project.  I haven't used it long enough to experience any problems, yet, but it appears to be the easiest way to start writing Cloudflare Workers.  It has the potential to be better than Miniflare, really.  Cloudflare will probably try to hire you, if you're no there already.  :)

marshallswain committed:
docs: add push guide

This adds a guide that walks through

- `denoflare push`
- script bindings in the .denoflare config
- publishing
- auto-publishing.

marshallswain commented:
I've also added a guide for `denoflare push` called "Publish a Worker".  It progressively walks through how to 

- Setup environment variable bindings in the `.denoflare` config.
- Reference a binding in the `envs` object.
- Use `denoflare push` to publish code.
- Use the `--name` option to explicitly provide a name.
- Use the `--watch` option to auto-publish

There's admittedly a small amount of overlap.  I simplified the previous guide to not show bindings in the config.  This way the reader is adding on top of what they've already learned.

![Screen Shot 2021-11-20 at 11 08 23 PM](https://user-images.githubusercontent.com/128857/142751720-1e4a9d4a-cdb0-4c30-bf30-b38b078b8c9f.jpg)

johnspurlock-skymethod reviewed (approved):
Excellent!  It's clear you've written good user guides before.  Happy to incorporate these into the project

johnspurlock-skymethod commented:
> Thanks for this project. I haven't used it long enough to experience any problems, yet, but it appears to be the easiest way to start writing Cloudflare Workers. It has the potential to be better than Miniflare, really. Cloudflare will probably try to hire you, if you're no there already. :)

Miniflare is great, and the [upcoming version of Wrangler](https://blog.cloudflare.com/wrangler-v2-beta/) looks like it will be much easier to do TS/esm workers as well.

They are both still Node based, though - and I think the Deno model maps much closer to the actual Workers production runtime and philosophy, so I'm excited to keep going with this Denoflare project as an option for folks that don't need/want npm/Nodeisms in their project.  I use it every day!

I'll probably always stay independent, the CF engineers are focused on building a great platform - and I'll be more than happy to build cool stuff on top of it

marshallswain commented:
I completely agree. It feels like it matches the values of building workers better.  So I'm looking forward to helping it grow where I can.  Thanks!
